### PR TITLE
Remove deprecated WorkerExecutor.submit

### LINKED ISSUE #21736: Remove @Deprecated DefaultWorkerExecutor methods

### PR DESCRIPTION
**Pull request by tresat**
- Many legacy tests which submitted Runnables seemed deletable

Fixes #21736

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle cancel

**Comment by tresat** [comment]
@bot-gradle test QFL

**Comment by tresat** [comment]
@bot-gradle test this

**Review by jvandort — APPROVED** [review]

File: subprojects/workers/src/integTest/groovy/org/gradle/workers/internal/WorkerExecutorCompositeBuildIntegrationTest.groovy [hunk 1]
```diff
@@ -237,11 +134,13 @@ class WorkerExecutorCompositeBuildIntegrationTest extends AbstractIntegrationSpe
 
                 @TaskAction
                 public void runWork() {
-                    workerExecutor.noIsolation().submit(TypedWorkAction.class, new Action<TypedParameter>() {
+                    WorkQueue queue = workerExecutor.noIsolation(s -> {});
```
Comment (big-guy) [inline subprojects/workers/src/integTest/groovy/org/gradle/workers/internal/WorkerExecutorCompositeBuildIntegrationTest.groovy:137]: There is a `noIsolation()` version of this

**Pull request merged**